src/structs/Webhook.ts

```diff
@@ -57,7 +59,33 @@ export class Webhook {
   }
 
   /**
-   * Middleware function to pass to express, sets req.vote to the payload
+   * Listening function for handling webhook requests
+   * @example
+   * app.post('/webhook', wh.listener((vote) => {
+   *   console.log(vote.user) // => 395526710101278721
+   * }))
+   * @param fn Vote handling function
+   * @returns An express request handler
+   */
+  public listener (fn: (payload: WebhookPayload, req?: Request, res?: Response, next?: NextFunction) => void | Promise<void>) {
+    return async (req: Request, res: Response, next: NextFunction) => {
+      const response = await this._parseRequest(req, res)
+      if (!response) return
+
+      res.sendStatus(204)
```

Comment:
We need to first check if a response was already sent in the user's function with
```suggestion
      if (!res.headersSent) {
        res.sendStatus(204)
      }
```
to make sure it doesn't cause issues with sending double headers. It might also be a good idea to make sure that `res.headersSent` is even a key in the response object in case this isn't being used in express directly.

Reply:
This requires other express properties/functions, do you think it's necessary to check the keys?